### ISSUE TITLE
Provide serialization api like serde_json

### ISSUE THREAD

**Issue by J-F-Liu** [alexcrichton/toml-rs#117]
Like

```
    let serialized = serde_json::to_string(&point).unwrap();
    let deserialized: Point = serde_json::from_str(&serialized).unwrap();
```

good to have

```
    let serialized = serde_toml::to_string(&point).unwrap();
    let deserialized: Point = serde_toml::from_str(&serialized).unwrap();
```

**Comment by alexcrichton** [comment]
This crate has `encode_str` and `decode_str`, and I think that covers cases like this?

**Comment by J-F-Liu** [comment]
`decode_str` needs
```
extern crate rustc_serialize;
#[derive(RustcDecodable)]
...
```

**Comment by alexcrichton** [comment]
I've rewritten the parser in the serde 0.9 rewrite: https://github.com/alexcrichton/toml-rs/pull/137 and I believe it should address this

**Comment by J-F-Liu** [comment]
Good, I'll test this when new crate release is available.

**Comment by alexcrichton** [comment]
Fixed in #137
